linera-core/src/worker.rs

```diff
@@ -121,6 +121,8 @@ pub enum Reason {
     NewBlock {
         height: BlockHeight,
         hash: CryptoHash,
+        timestamp: Timestamp,
+        new_chains: Vec<(ChainId, PublicKey)>,
```

Comment:
Good point, I guess it should be `Vec<(ChainId, ChainOwnership)>`.

You're right, I see one potential problem: Let's say a block A5 on chain A created chain B.

* If I receive the notification about A5 first, I start tracking chain B and subscribe to notifications for it. So I will receive all of those _after_ the notification about A5, which is good. :heavy_check_mark: 
* But if I for some reason subscribe to notifications about B first, and receive a `NewBlock` about, say, B2, then if I _later_ get the `NewBlock` about A5, I would currently replace the entry for B with a new one (next height 0 instead of 3). :x: 

I think I just need to make sure `update_wallet_for_new_chain` never overwrites existing chains.

Reply:
Ah, the other thing is something I haven't even implemented (yet):

If ownership changes and we _stop_ being a chain owner, do we want to stop tracking it?

Reply:
…might as well replace `public_key` in `OpenChain` with `ChainOwnership`, too?

Reply:
sounds good to me

Reply:
As discussed, I'll do that in a separate PR.